applications/plugins/SofaMatrix/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+cmake_minimum_required(VERSION 3.12)
+project(SofaMatrix VERSION 1.0 LANGUAGES CXX)
+
+sofa_find_package(SofaBaseLinearSolver REQUIRED)
+sofa_find_package(CImgPlugin REQUIRED)
+sofa_find_package(SofaGuiQt REQUIRED QUIET)
+
+set(SOFA_MODULES
+    SofaBaseLinearSolver
+    CImgPlugin
+)
+
+set(SOFAMATRIX_SRC_DIR src/${PROJECT_NAME})
+
+set(HEADER_FILES
+    ${SOFAMATRIX_SRC_DIR}/config.h.in
+    ${SOFAMATRIX_SRC_DIR}/MatrixImageExporter.h
+)
+set(SOURCE_FILES
+    ${SOFAMATRIX_SRC_DIR}/initSofaMatrix.cpp
+    ${SOFAMATRIX_SRC_DIR}/MatrixImageExporter.cpp
+)
+set(README_FILES
+    SofaMatrix.md
+)
+set(MOC_FILES
+)
+set(MOC_HEADER_FILES
+)
+
+if (NOT SofaGuiQt_FOUND)
+    message(NOTICE "[SofaMatrix] Module SofaGuiQt not found: some components (GlobalSystemMatrixImage) will not be compiled")
+else()
+    list(APPEND HEADER_FILES
+        ${SOFAMATRIX_SRC_DIR}/GlobalSystemMatrixImage.h
+        ${SOFAMATRIX_SRC_DIR}/BaseMatrixImageProxy.h
+    )
+    list(APPEND SOURCE_FILES
+        ${SOFAMATRIX_SRC_DIR}/GlobalSystemMatrixImage.cpp
+        ${SOFAMATRIX_SRC_DIR}/BaseMatrixImageViewerWidget.cpp
+    )
+    list(APPEND MOC_HEADER_FILES
+        ${SOFAMATRIX_SRC_DIR}/BaseMatrixImageViewerWidget.h
+    )
+    list(APPEND SOFA_MODULES
+        SofaGuiQt
+    )
+
+    find_package(Qt5 COMPONENTS Core Gui Widgets QUIET)
+    if(Qt5Core_FOUND)
+        qt5_wrap_cpp(MOC_FILES ${MOC_HEADER_FILES})
+    else()
+        find_package(Qt4 COMPONENTS qtcore)
```

Comment:
... you want to support qt4 ? 😵
I would advise to focus more on qt6 🧐

Reply:
you are right. It is copy/pasted from the `image` plugin.